### ISSUE TITLE
expand support to new/multiple backends

### ISSUE THREAD

**Issue by beckermr** [biggles-plot/biggles#43]
plotutils does not seem to be actively maintained at the moment. Thus I think we should try and allow for plotting with a different backend. @esheldon suggested Cairo (http://cairographics.org/).

**Comment by beckermr** [comment]
I have been reading the Cairo docs. This should be possible. The xwindows event loops seem a touch tricky to get right and not buggy.

**Comment by esheldon** [comment]
another alternative backend that looks interesting is pyx.  I think it can do everything biggles does, but has clunky syntax.  It is pure python.  Would just need to wrap everything in nice classes.  For the show() command would need to spawn to an external program.

**Comment by beckermr** [comment]
Yeah. I ran into that. Cairo has pycario which should put everything in Python too. I will look again.

**Comment by beckermr** [comment]
I have been reading the pyx docs and their support for transparency seems limited. 

http://pyx.sourceforge.net/manual/color.html#transparency

This is the major motivation for me for moving to a new backend. 

Thus I think cairo is probably a better choice since I think it supports this well.

**Comment by esheldon** [comment]
if we are willing to write pdf then convert to png (and then possibly display that on the screen) then pyx could work.

**Comment by beckermr** [comment]
This all sounds clunky and slow. I am thinking supporting both backends (plotutils and Cairo+gtk) is the way to go. If you don't have gtk, you can revert back to plotutils. Thoughts?

**Comment by esheldon** [comment]
I think cairo is definitely worth looking into.

Also I looked into it, and I think I could hack together a barebones X11 image display that would be as fast as plotutils current one (gtk is unbearably slow over the network)

**Comment by beckermr** [comment]
That sounds good. Cairo's docs seem to imply that it really wants you to use gtk. But hopefully this is not a strict requirement.

**Comment by esheldon** [comment]
I think the idea is to have cairo write directly to some display buffer, which is I think what plotutils does.
